crates/bevy_math/src/linear_convex_space.rs

```diff
@@ -0,0 +1,69 @@
+//! This module provides the `LinearConvexSpace` trait and some related objects
+use glam::{DVec2, DVec3, DVec4, Quat, UVec2, UVec3, UVec4, Vec2, Vec3, Vec3A, Vec4};
+
+/// A trait providing methods for spanning a linear, convex space.
+///
+/// A convex space or convex set is a set of points (think points in 3d or 2d space) in which every point on a line between two points A and B in that set
+/// is also a part of that set. For example the set of all points inside a square is a convex set as you can draw a line between any two points in the square and never leave it.
+/// For more information, please see [this wikipedia article](https://en.wikipedia.org/wiki/Convex_set).
+///
+/// The space should also be linear, meaning that `A*t + B*(1 - t)` with any points `A`, `B` and `t` element `0..1` should represent a straight line.
+///
+/// By implementing this trait, you guarantee that the above conditions hold true.
+pub trait LinearConvexSpace<Scalar>: Default + Copy + Clone {
+    /// Adds two elements of the space.
```

Comment:
I think, instead of redefining the standard operators, we should provide the methods that are directly used by splines as primitives here.

That would be:

- Scale (vector * scalar -> vector), and provide a method with default impl for a `scale_recip` which does the scalar division
- Dot product (vector * vector -> scalar), and use it to perform the matrix multiplication

It would still simplify the code of users of `LinearConvexSpace` while also not making it look like we're just reimplementing the standard operators (which, theoretically, we are not, we're only providing an implementation of a vector space that is not necessarily linear, but has

Reply:
I agree that `mul` and `div` could (and should) be replaced with `scale` and `scale_recip` but I don't see where the dot product would be used here. AFAIK we need some kind of "add" (and sub) but a dot product on `P: LinearConvexSpace` is currently neither possible nor needed.

Reply:
Ah I misread the original implementation and convinced myself the matrix multiplication could be done with dot products 🤦🏻 

That makes it a bit impossible to make the implementations nicer on the eyes, which is disappointing to me.

Reply:
No worries :) We could still make it nicer using a wrapper type similar to what was suggested [in this comment](https://github.com/bevyengine/bevy/pull/12534#discussion_r1527618411) but I would defer this to a future PR aswell 😅